components/automate-cli/pkg/verifyserver/services/sshusercheckservice/sshusercheckservice.go

```diff
@@ -0,0 +1,126 @@
+package sshusercheckservice
+
+import (
+	"fmt"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/io/fileutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/chef/automate/lib/sshutils"
+)
+
+type SshUserCheckService interface {
+	CheckSshUserDetails(*models.SshUserChecksRequest) (*models.SshUserChecksResponse, error)
+}
+
+type SshUserServiceImpl struct {
+	logger      logger.Logger
+	sshUtil     sshutils.SSHUtil
+	pemFileName string
+	fileutils   fileutils.FileUtils
+}
+
+func NewSshUserCheckService(logger logger.Logger, fileutils fileutils.FileUtils, sshUtils sshutils.SSHUtil) *SshUserServiceImpl {
+	return &SshUserServiceImpl{
+		logger:      logger,
+		sshUtil:     sshUtils,
+		pemFileName: PEM_FILE_NAME,
+		fileutils:   fileutils,
+	}
+}
+
+const (
+	SUDO_PASSWORD_CMD = `echo "%s" | sudo -S ls -l`
+	PEM_FILE_NAME     = "private_key"
+)
+
+func (ssu *SshUserServiceImpl) CheckSshUserDetails(req *models.SshUserChecksRequest) (*models.SshUserChecksResponse, error) {
+	ssu.logger.Debug("The Request Value for the SSH User Check: ", req)
+	serviceResponseArray := []*models.Checks{}
+	serviceResponse := &models.SshUserChecksResponse{}
+	serviceResponse.Passed = true
+
+	filePath, sshConfig, err := ssu.getConfig(req.Ip, req.Private_Key, req.User_Name, req.Port, req.Sudo_Password)
```

Comment:
You can pass entire object instead of passing individually

Reply:
Done